src/domain/collection/curatedEventList/CuratedEventList.tsx

```diff
@@ -25,18 +30,60 @@ const CuratedEventList: React.FC<Props> = ({ collection }) => {
   const { t } = useTranslation();
   const locale = useLocale();
   const [showAllPastEvents, setShowAllPastEvents] = React.useState(false);
+  const [isFetchingMore, setIsFetchingMore] = React.useState(false);
+  const page = React.useRef(INITIAL_PAGE);
+  const currentEventsCount = page.current * PAGE_SIZE;
   const eventIds = React.useMemo(
     () =>
       collection.curatedEvents
         .map((url) => getEventIdFromUrl(url) as string)
         .filter((e) => e),
     [collection.curatedEvents]
   );
+  const isMoreToLoad = currentEventsCount < eventIds.length;
 
-  const { data: eventsData, loading } = useEventsByIdsQuery({
-    variables: { ids: eventIds, include: ['keywords', 'location'] },
+  const queryVariables = {
+    ids: eventIds.slice(0, PAGE_SIZE),
+    include: ['keywords', 'location'],
+  };
+
+  const {
+    data: eventsData,
+    loading,
+    fetchMore,
+    client: apolloClient,
+  } = useEventsByIdsQuery({
+    variables: queryVariables,
   });
 
+  // When component is unmounting, we need to delete the paginated data from cache
+  // and only persist the first page with writeQuery. Otherwise when page count resets and
+  // events are fetched again, it returns all the events form cache that were previously fetched.
+  // Then page and event array lenght don't match up...
+  React.useEffect(() => {
+    return () => {
+      try {
+        const data = apolloClient.readQuery({
+          query: EventsByIdsDocument,
+          variables: queryVariables,
+        });
+        apolloClient.writeQuery({
+          query: EventsByIdsDocument,
+          variables: queryVariables,
+          data: {
+            eventsByIds: data.eventsByIds.slice(0, PAGE_SIZE),
+          },
+        });
+      } catch (e) {
+        if (process.env.NODE_ENV !== 'test') {
+          // eslint-disable-next-line no-console
+          console.error('Clearing cache failed: ', e);
+        }
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
This part was little bit challenging. Need to clear cache to keep pagination workiong when going away from the page and coming back. If someone has better idea how to handle this, pleas tell :)

Reply:
I really dont know much about apollo cache, but when quickly reading docs I came across with connection direct, could this help with the problem: https://www.apollographql.com/docs/react/caching/advanced-topics/#the-connection-directive ?

Reply:
Also OtherEventTimes.tsx uses React.useCallback solution which is another option? https://github.com/City-of-Helsinki/events-helsinki-ui/blob/master/src/domain/event/eventInfo/OtherEventTimes.tsx#L55

Reply:
Interesting... it seems that otherEventTimes do not even use load more button, but loads more when scrolling. I think it is pretty much the same as this collection event list in that sense that other events are quite fixed value and rarely updated.

Reply:
For example here: https://tapahtumat.test.kuva.hel.ninja/fi/event/helsinki:af32joap6m
![Screenshot 2021-02-15 at 8 40 06 PM](https://user-images.githubusercontent.com/7648194/107983476-83327400-6fce-11eb-8f05-ecc9560b92f3.png)

Reply:
Yeah this loads all the pages:

```
  React.useEffect(() => {
    const page = subEventsData?.eventList.meta
      ? getNextPage(subEventsData.eventList.meta)
      : null;

    if (page) {
      handleLoadMore(page);
    }
  }, [handleLoadMore, subEventsData]);
```